scope.go

```diff
@@ -281,6 +286,39 @@ func (s *scope) reportRegistry() {
 	}
 }
 
+func (s *scope) processLoop(interval time.Duration) {
+	ticker := time.NewTicker(interval)
+	counters := make([]*counter, 0, _defaultInitialSliceSize)
+	gauges := make([]*gauge, 0, _defaultInitialSliceSize)
+
+	defer ticker.Stop()
+	for {
+		select {
+		case c := <-s.counterChangeNotifyCh:
+			counters = append(counters, c)
```

Comment:
what happens if `c` is part of a subscope that's closed before you process it?

Reply:
As discussed, this will have the effect of flushing out any pending accumulated metrics which is desirable.